scripts/users.js

```diff
@@ -0,0 +1,275 @@
+  let nameError = document.getElementById("nameError");
+  let emailError = document.getElementById("emailError");
+  let ageError = document.getElementById("ageError");
+  let addressError = document.getElementById("addressError");
+
+  if (
+    !nameRegex.test(name) ||
+    !emailRegex.test(email) ||
+    age < 0 ||
+    age > 150 ||
+    !addressRegex.test(address)
+  ) {
+    if (!nameRegex.test(name)) {
+      nameError.innerHTML =
+        "Name should be between 1-30 characters, uppercase and lowercase letters and space are allowed.";
+    } else {
+      nameError.innerHTML = "";
+    }
+    if (!emailRegex.test(email)) {
+      emailError.innerHTML = "Email should be valid.";
+    } else{
+      emailError.innerHTML = "";
+    }
+    if (age < 0 || age > 150 || age.toString() === "") {
+      ageError.innerHTML = "Age should be between 0-150.";
+    } else{
+      ageError.innerHTML = "";
+    }
+    if (!addressRegex.test(address)) {
+      addressError.innerHTML =
+        "Address should be betweeen 1-300 characters and uppercase, lowercase, numbers, space are allowed.";
+    }else{
+      addressError.innerHTML = "";
+    }
+  } else {
+    nameError.innerHTML = "";
+    emailError.innerHTML = "";
+    ageError.innerHTML = "";
+    addressError.innerHTML = "";
+
+    let user = {
+      id: id || newUserId,
+      name,
+      email,
+      age,
+      address,
+    };
+
+    if (id) {
+      editUser(user);
+    } else {
+      addUserToTable(user);
+      addUserToJSON(user);
+    }
+
+    form.reset();
+  }
+});
+
+const loadUserToForm = (user) => {
+  //Loads the data from the local storage, because the form doesn't update until the page is refreshed.
+  let userData = getUsers().find((u) => u.id === user.id);
+
+  form.id.value = sanitize(userData.id);
+  form.name.value = sanitize(userData.name);
+  form.email.value = sanitize(userData.email);
+  form.age.value = sanitize(userData.age.toString());
+  form.address.value = sanitize(userData.address);
+
+  form.scrollIntoView();
+};
+
+const editUser = (user) => {
+  let database = getDatabase();
+  let users = getUsers();
+  let userRow = document.getElementById(user.id).children;
+
+  let userIndex = users.findIndex((u) => {
+    return user.id === u.id;
+  });
+
+  if (userIndex > -1) {
+    users[userIndex] = user;
+    localStorage.setItem(
+      "database",
+      JSON.stringify({
+        ...database,
+        users,
+      })
+    );
+
+    userRow[1].innerHTML = sanitize(user.name);
+    userRow[2].innerHTML = sanitize(user.email);
+    userRow[3].innerHTML = sanitize(user.age.toString());
+    userRow[4].innerHTML = sanitize(user.address);
+  } else {
+    alert("No user found.");
+  }
+};
+
+const deleteUser = (user) => {
+  let deleteModal = confirm(`Do you want to delete user ${user.name}?`);
+
+  if (deleteModal) {
+    tableBody.removeChild(document.getElementById(user.id));
+
+    let users = getUsers().filter((u) => u.id !== user.id);
+    localStorage.setItem(
```

Comment:
It's better to have reusable functions that encapsulate the access to the localStorage. You already have some `getUsers` and `getDatabase` but you don't fully reuse them and also you need to extend the API so that nowhere in the code invoke the undelying localStorage API directly.
The point for this is to hide the dependency and put it under control so that it's managed in one place. This way it'd be easy to change and extend it or even replace the local storage with real DB for example.